File: src/main/java/org/shark/example/controller/file/FileController.java
package org.shark.example.controller.file;

import lombok.RequiredArgsConstructor;
import org.shark.example.service.file.SaveFileApiService;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestPart;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.multipart.MultipartFile;

@RestController
@RequestMapping("file")
@RequiredArgsConstructor
public class FileController {

    private final SaveFileApiService saveFileApiService;

    @PostMapping(consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
    public void save(@RequestPart MultipartFile file) {
        saveFileApiService.start(file);
    }
}

File: src/main/kotlin/org/shark/example/controller/file/FileController.kt
package org.shark.example.controller.file

import org.shark.example.service.file.SaveFileApiService
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestPart
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.multipart.MultipartFile

@RestController
@RequestMapping("file")
class FileController(
    private val saveFileApiService: SaveFileApiService
) {

    @PostMapping(consumes = [MediaType.MULTIPART_FORM_DATA_VALUE])
    fun save(@RequestPart file: MultipartFile) {
        saveFileApiService.start(file)
    }
}
